packages/nimble-components/src/wafer-map/modules/experimental/worker-renderer.ts

```diff
@@ -1,14 +1,82 @@
+import { type Remote, transfer } from 'comlink';
 import type { WaferMap } from '../..';
 import { HoverDieOpacity } from '../../types';
+import { createMatrixRenderer } from '../create-matrix-renderer';
+import type { MatrixRenderer } from '../../../../build/generate-workers/dist/esm/source/matrix-renderer';
 
 /**
  * Responsible for drawing the dies inside the wafer map, adding dieText and scaling the canvas
  */
 export class WorkerRenderer {
+    private matrixRenderer!: Remote<MatrixRenderer>;
+
     public constructor(private readonly wafermap: WaferMap) {}
 
-    public drawWafer(): void {
-        // rendering will be implemented in a future PR
+    public async setupWafer(): Promise<void> {
+        if (this.matrixRenderer === undefined) {
+            const { matrixRenderer } = await createMatrixRenderer();
+            this.matrixRenderer = matrixRenderer;
+            const offscreenCanvas = this.wafermap.workerCanvas.transferControlToOffscreen();
+            await this.matrixRenderer.setCanvas(
+                transfer(offscreenCanvas, [offscreenCanvas])
+            );
+        }
+
+        await this.matrixRenderer.setCanvasDimensions({
+            width: this.wafermap.canvasWidth ?? 0,
+            height: this.wafermap.canvasHeight ?? 0
+        });
+        await this.matrixRenderer.setDiesDimensions(
+            this.wafermap.experimentalDataManager.dieDimensions
+        );
+        const scaleX = this.wafermap.experimentalDataManager.horizontalScale(1)!
+            - this.wafermap.experimentalDataManager.horizontalScale(0)!;
+        const scaleY = this.wafermap.experimentalDataManager.verticalScale(1)!
+            - this.wafermap.experimentalDataManager.verticalScale(0)!;
+        await this.matrixRenderer.setScaling(scaleX, scaleY);
+        await this.matrixRenderer.setBases(
+            this.wafermap.experimentalDataManager.horizontalScale(0)!,
+            this.wafermap.experimentalDataManager.verticalScale(0)!
+        );
+        await this.matrixRenderer.setMargin(
+            this.wafermap.experimentalDataManager.margin
+        );
+
+        if (this.wafermap.diesTable !== undefined) {
```

Comment:
I think all these checks for diesTable are undefined are incorrect because they result in the worker not receiving any message.

But the user could intentionally be trying to clear the current wafer that is being drawn. Now if you set diesTable to undefined nothing happens right (the wafer canvas stays the same even though the data was cleared)?

Maybe we are being too clever with the diesTable property and renderer. We should consider adding a boolean property like `use-experimental-renderer` and just be explicit for the user to opt-in. 

I think there are two outcomes and they should not block this PR:
- setting and clearing the diesTable should behave well
- (up to y'all but I think it will help) we should make an explicit boolean configuration to switch rendering of the wafer map.

If that makes sense can you track it under `Render Dies` in the tracking issue? https://github.com/ni/nimble/issues/2034

Reply:
So we have a flag on the wafer-map called isExperimentalRenderer() but it is checked before calling setupWafer(), inside update() from index.ts there is a 
`
        if (this.isExperimentalUpdate()) {
            this.currentTask = this.experimentalUpdate();
            return;
        }
`
and the call stack is update() -> this.experimentalUpdate() -> this.workerRenderer.setupWafer();

so we can assume every time this.workerRenderer.setupWafer it is an experimental wafer

Regarding the undefined checking and the worker not receiveing messages, I replaced that check with 
![image](https://github.com/ni/nimble/assets/110180309/1ab3f044-a8ae-45c7-bf7d-e22a04afdc9e)
now the wafer is cleared when diesTable is undefined

Reply:
Reopening this question. @Razvan1928 

- isExperimentalUpdate just checks that diesTable is undefined, are you sure that is sufficient? Did you do any manual verification?
- are there tests to verify the behavior change? Ie the wafer if drawn, then the data is changed to undefined, and the wafer is cleared?

If there are not tests there needs to be a follow-up PR